Softeq.XToolkit.PushNotifications.iOS/IosPushNotificationsService.cs

```
@@ -0,0 +1,122 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using Softeq.XToolkit.Common.Interfaces;
+using UIKit;
+using UserNotifications;
+
+namespace Softeq.XToolkit.PushNotifications.iOS
+{
+    public class IosPushNotificationsService : PushNotificationsServiceBase
+    {
+        private readonly INotificationsPermissionsService _permissionsService;
+
+        private bool _isInitialized;
+
+        // badges
+
+        public IosPushNotificationsService(IRemotePushNotificationsService remotePushNotificationsService,
+            IPushTokenStorageService pushTokenStorageService, IPushNotificationsHandler pushNotificationsHandler,
+            IPushNotificationParser pushNotificationParser, INotificationsPermissionsService permissionsService, ILogManager logManager)
+            : base(remotePushNotificationsService, pushTokenStorageService, pushNotificationsHandler, pushNotificationParser, logManager)
+        {
+            _permissionsService = permissionsService;
+        }
+
+        public override void Initialize(bool showForegroundNotificationsInSystem)
+        {
+            if (_isInitialized)
+            {
+                Logger.Debug("PushNotificationsServiceIos: Already Initialized");
```

should we throw an exception?

---

fixed